static/js/apps/eval_retrieval_generation/query_section.tsx

```diff
@@ -104,6 +105,58 @@ function getAnswer(
     });
 }
 
+/**
+ * Absolutely positions the tooltip within the given annotation so it fits
+ * within the parent query section.
+ */
+function adjustTooltipPosition(annotationEl: Element): void {
+  const tooltipEl = annotationEl.querySelector(
+    ".dc-stat-tooltip"
+  ) as HTMLDivElement;
+  const parentSection = annotationEl.closest("#query-section");
+  const sectionRect = parentSection.getBoundingClientRect();
+
+  // Limit tooltip width to section width.
+  if (tooltipEl.getBoundingClientRect().width > sectionRect.width - 2) {
+    tooltipEl.style.maxWidth = `${sectionRect.width - 2}px`;
+  }
+
+  // Re-calculate since tooltip width may have changed.
+  const tooltipRect = tooltipEl.getBoundingClientRect();
+  const tooltipWidth = tooltipRect.width;
+
+  // All tooltip positioning is relative to the annotation el's left edge.
+  const annotationRect = annotationEl.getBoundingClientRect();
+  // With CSS transform attribute, tooltip is shifted by this much.
```

Comment:
nit: I wonder if it'll be simpler to do all of the tooltip positioning here instead of having tooltip positioning in both CSS and here. Here we know the left coordinate of both the section and the annotation so can set the left of the tooltip accordingly?

Reply:
That makes sense to me! Updated.